native/src/main/java/io/ballerina/stdlib/http/api/HttpService.java

```diff
@@ -258,9 +262,108 @@ private static void processResources(HttpService httpService) {
+        }
+        for (HttpResource resource : resources) {
+            if (resource.getResourceName() == null) {
+                continue;
+            }
+            if (targetResource.getResourceName().equalsIgnoreCase(resource.getResourceName()) &&
+                    !targetResource.getResourcePathSignature().equals(resource.getResourcePathSignature())) {
+                throw new BallerinaConnectorException("cannot duplicate resource name: '" +
+                                                      targetResource.getResourceName() + "' unless they " +
+                                                      "have the same path");
+            }
+        }
+    }
+
+    private static void setLinkToResource(HttpService httpService, HttpResource targetResource,
+                                          HttpResource.LinkedResource link, HttpResource linkedResource) {
+        BMap<BString, Object> linkMap = ValueCreatorUtils.createHTTPRecordValue(HttpConstants.LINK);
+        BString relation = fromString(link.getRelationship());
+        validateLinkRelationUniqueness(targetResource, relation);
+        linkMap.put(HttpConstants.LINK_REL, relation);
+        BString href = fromString(linkedResource.getAbsoluteResourcePath());
+        linkMap.put(HttpConstants.LINK_HREF, href);
+        BArray methods = getMethodsBArray(linkedResource.getMethods());
+        linkMap.put(HttpConstants.LINK_METHODS, methods);
+        String returnMediaType = linkedResource.getReturnMediaType();
+        if (httpService.getMediaTypeSubtypePrefix() != null) {
+            String specificReturnMediaType = HttpUtil.getMediaTypeWithPrefix(httpService.getMediaTypeSubtypePrefix(),
+                                                                             returnMediaType);
+            if (specificReturnMediaType != null) {
+                returnMediaType = specificReturnMediaType;
+            }
+        }
+        if (returnMediaType != null) {
+            BArray types = ValueCreator.createArrayValue(TypeCreator.createArrayType(PredefinedTypes.TYPE_STRING));
+            types.append(fromString(returnMediaType));
+            linkMap.put(HttpConstants.LINK_TYPES, types);
+        }
+        targetResource.addLink(relation, linkMap);
+    }
+
+    private static BArray getMethodsBArray(List<String> methods) {
+        if (methods == null) {
+            methods = Arrays.asList("GET", "POST", "PUT", "PATCH", "DELETE", "OPTIONS", "HEAD");
+        }
+        List<BString> methodsAsBString = methods.stream().map(StringUtils::fromString).collect(Collectors.toList());
+        return ValueCreator.createArrayValue(methodsAsBString.toArray(BString[]::new));
+    }
+
+    private static void validateLinkRelationUniqueness(HttpResource targetResource, BString relation) {
+        if (targetResource.hasLinkedRelation(relation)) {
+            throw new BallerinaConnectorException("cannot duplicate resource relation: '" + relation.getValue() + "'");
```

Comment:
Better to throw BError if possible, we need to deprecate the use of `BallerinaConnectorException`

Reply:
Fixed in [9b1a15c](https://github.com/ballerina-platform/module-ballerina-http/pull/1068/commits/9b1a15ce791c2179473d67578d1acbb8c934d313)